### PR TITLE
Added numpy for GPU nodes

### PR DESCRIPTION
**Pull request by nvarini**

### CODE REVIEW

**Comment by rmsds** [comment]
Sorry for the back and forth, my previous commit actually made py-torch also be installed for python 3 (which we don't want afaik).

File: paien.yaml [hunk 1]
```diff
@@ -612,6 +612,18 @@ packages:
...
+  gnu-gpu-lapack-python-libraries:
+    target_matrix:
+      - gnu-gpu
+    requires:
+      - architecture
+      - compiler
+      - python
+      - lapack
+    specs:
+      - py-numpy+blas+lapack@1.14.3
```
Comment (alalazo) [inline paien.yaml:626]: Do we want this or:
```
py-numpy+blas+lapack@1.14.3 ^magma@2.3.0+fortran+shared ^cuda@9.1.85
```
on GPU nodes?

**Review by alalazo — CHANGES_REQUESTED** [review]
I have a comment on how `py-numpy` gets built on GPU nodes. Can anybody give more information on that?

File: paien.yaml (hunk 1 above)
Comment (rmsds) [reply paien.yaml:626]: It seems to me that py-numpy does not depend on magma. It's py-pytorch (a few lines above) which depends on magma (which depends on cuda), which is just above py-pytorch in the list.

Comment (alalazo) [reply paien.yaml:626]: @rmsds Double checked, I think you're right. I thought `magma` was a `lapack` provider for GPUs, but it is not apparently. Then this is fine with me. I'll approve feel free to merge. Otoh `py-numpy` should have been activated along with `py-torch`.

**Review by alalazo — APPROVED** [review]

**Pull request merged**